### DISCUSSION TITLE
9P Filesystem Alternative

### DISCUSSION THREAD

**Discussion by cemalgnlts** [copy/v86#862] [Q&A] [answered]
Hi,

Is there a way to read/write files in the emulator other than the 9p filesystem?

**Comment by ProgrammerIn-wonderland** [comment]
curious, what's your usecase? why do you not want to use 9pfs?

**Reply by cemalgnlts** [reply]
I don't remember exactly what I was doing at the time, but I couldn't install 9PFS for Tiny Linux so I was looking for an alternative.

**Reply by SuperMaxusa** [reply]
> but I couldn't install 9PFS for Tiny Linux

You need get a linux source files and patches ([15.x](http://tinycorelinux.net/15.x/x86/release/src/kernel/), [14.x](http://tinycorelinux.net/14.x/x86/release/src/kernel/)) for Tiny Core and recompile kernel with these enabled modules in kernel configuration:
https://github.com/copy/v86/blob/2bcfa9d8c427f5b1d395b8ff931751550ce25e42/docs/linux-9p-image.md?plain=1#L4-L11
https://www.parkytowers.me.uk/thin/Linux/TinycoreCK.shtml

p.s. On Tiny Core Linux 15.x (6.6.8 kernel) 9pfs may not properly work on creating files due this issue: #996

**Reply by cemalgnlts** [reply]
Thanks, I will share the result if I try it one day.

**Reply by SuperMaxusa** [reply]
For interest I tried to build a kernel for Tiny Core Linux 15.0 with 9PFS support: [tinycore9p.tar.gz](https://github.com/user-attachments/files/16325263/tinycore9p.tar.gz)
You need replace `vmlinuz` from the archive on your current installation or in LiveCD, not yet tested whether FSCache works or not, `.config` is available too.

![9pfs-tcl](https://github.com/user-attachments/assets/51a243d1-4056-4010-9e8a-529556285c0a)

**Comment by SuperMaxusa** [comment, marked as answer]
see #824 and https://github.com/copy/v86/issues/965#issuecomment-1880384399
